manifests/virtualhost.pp

```diff
@@ -1,14 +1,15 @@
 # == Type: prosody::virtualhost
 define prosody::virtualhost (
-  Hash                           $custom_options = {},
-  Enum[present, absent]          $ensure         = present,
-  Optional[Stdlib::Absolutepath] $ssl_key        = undef,
-  Optional[Stdlib::Absolutepath] $ssl_cert       = undef,
-  Boolean                        $ssl_copy       = true,
-  Optional[String]               $user           = undef,
-  Optional[String]               $group          = undef,
-  Hash                           $components     = {},
-  Array[String[1]]               $disco_items    = [],
+  Hash                           $custom_options   = {},
+  Enum[present, absent]          $ensure           = present,
+  Optional[Stdlib::Absolutepath] $ssl_key          = undef,
+  Optional[Stdlib::Absolutepath] $ssl_cert         = undef,
+  Boolean                        $ssl_copy         = true,
+  Optional[String]               $user             = lookup('prosody::user'),
+  Optional[String]               $group            = lookup('prosody::group'),
+  Hash                           $components       = {},
+  Array[String[1]]               $disco_items      = [],
+  Stdlib::Absolutepath           $config_directory = lookup('prosody::config::config_directory'),
```

Comment:
Same as above for this parameter: remove it, include `prosody` in the beginning to be sure `$prosody::config_directory` can be used and use this where applicable.